### ISSUE TITLE
HTTP 500 Error on API call

### ISSUE THREAD

**Issue by subesokun** [spotify/luigi#2154]
Hi,

I know there have been already a bunch of issues related to this particular HTTP 500 Error but they've been closed after the version range bugfix for `tornado` has been release. Sadly I'm still getting the 500 errors as shown below in the log.

Note: As I'm running the `luigid` within a Docker container I can't use the `--background` flag as otherwise the container will terminate. But not sure if this is related to this issue as the Web UI works fine and `luigid` is running.

Any ideas?

```
daemon_1  | Configuring logging from file: /etc/luigi/logging.conf
daemon_1  | 2017-06-12 14:28:10,308 tornado.application ERROR    Uncaught exception GET /api/ping (192.168.0.1)
daemon_1  | HTTPServerRequest(protocol='http', host='some-host:8082', method='GET', uri='/api/ping', version='HTTP/1.1', remote_ip='192.168.0.1', headers={'Host': 'some-host:8082', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'User-Agent': 'python-requests/2.14.2', 'Content-Length': '0', 'Content-Type': 'application/x-www-form-urlencoded', 'Cache-Control': 'max-stale=0', 'Connection': 'Keep-Alive', 'X-Bluecoat-Via': 'dr56cedb74de417a'})
daemon_1  | Traceback (most recent call last):
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/tornado/web.py", line 1509, in _execute
daemon_1  |     result = method(*self.path_args, **self.path_kwargs)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/server.py", line 77, in get
daemon_1  |     result = getattr(self._scheduler, method)(**arguments)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/scheduler.py", line 1151, in ping
daemon_1  |     worker_id = kwargs['worker']
daemon_1  | KeyError: 'worker'
daemon_1  | 2017-06-12 14:28:10,317 tornado.access  ERROR    500 GET /api/ping (192.168.0.1) 9.75ms
daemon_1  | 2017-06-12 14:28:15,146 tornado.application ERROR    Uncaught exception GET /api/add_task (192.168.0.1)
daemon_1  | HTTPServerRequest(protocol='http', host='some-host:8082', method='GET', uri='/api/add_task', version='HTTP/1.1', remote_ip='192.168.0.1', headers={'Host': 'some-host:8082', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'User-Agent': 'python-requests/2.14.2', 'Content-Length': '0', 'Content-Type': 'application/x-www-form-urlencoded', 'Cache-Control': 'max-stale=0', 'Connection': 'Keep-Alive', 'X-Bluecoat-Via': 'dr56cedb74de417a'})
daemon_1  | Traceback (most recent call last):
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/tornado/web.py", line 1509, in _execute
daemon_1  |     result = method(*self.path_args, **self.path_kwargs)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/server.py", line 77, in get
daemon_1  |     result = getattr(self._scheduler, method)(**arguments)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/scheduler.py", line 772, in add_task
daemon_1  |     assert worker is not None
daemon_1  | AssertionError
daemon_1  | 2017-06-12 14:28:15,148 tornado.access  ERROR    500 GET /api/add_task (192.168.0.1) 2.41ms
daemon_1  | 2017-06-12 14:28:40,363 tornado.application ERROR    Uncaught exception GET /api/ping (192.168.0.1)
daemon_1  | HTTPServerRequest(protocol='http', host='some-host:8082', method='GET', uri='/api/ping', version='HTTP/1.1', remote_ip='192.168.0.1', headers={'Host': 'some-host:8082', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'User-Agent': 'python-requests/2.14.2', 'Content-Length': '0', 'Content-Type': 'application/x-www-form-urlencoded', 'Cache-Control': 'max-stale=0', 'Connection': 'Keep-Alive', 'X-Bluecoat-Via': 'dr56cedb74de417a'})
daemon_1  | Traceback (most recent call last):
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/tornado/web.py", line 1509, in _execute
daemon_1  |     result = method(*self.path_args, **self.path_kwargs)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/server.py", line 77, in get
daemon_1  |     result = getattr(self._scheduler, method)(**arguments)
daemon_1  |   File "/usr/local/lib/python3.6/site-packages/luigi/scheduler.py", line 1151, in ping
daemon_1  |     worker_id = kwargs['worker']
daemon_1  | KeyError: 'worker'
```

Server configuration:
```
> pip list
docutils (0.13.1)
lockfile (0.12.2)
luigi (2.6.2)
pip (9.0.1)
python-daemon (2.1.2)
setuptools (36.0.1)
SQLAlchemy (1.1.10)
tornado (4.5.1)
wheel (0.29.0)
```

Client configuration:
```
> conda list
asn1crypto                0.22.0                   py35_0
cffi                      1.10.0                   py35_0
conda                     4.2.13                   py35_0    conda-forge
conda-env                 2.6.0                         0    conda-forge
cryptography              1.8.1                    py35_0
idna                      2.5                      py35_0
libffi                    3.2.1                         1
lockfile                  0.12.2                   py35_0    conda-forge
luigi                     2.6.2                    py35_0    conda-forge
openssl                   1.0.2l                        0
packaging                 16.8                     py35_0
pip                       9.0.1                    py35_1
pycosat                   0.6.2                    py35_0
pycparser                 2.17                     py35_0
pyopenssl                 17.0.0                   py35_0
pyparsing                 2.1.4                    py35_0
python                    3.5.3                         1
python-daemon             2.1.2                    py35_0
readline                  6.2                           2
requests                  2.14.2                   py35_0
ruamel_yaml               0.11.14                  py35_1
setuptools                27.2.0                   py35_0
six                       1.10.0                   py35_0
sqlite                    3.13.0                        0
tk                        8.5.18                        0
tornado                   4.5.1                    py35_0    conda-forge
wheel                     0.29.0                   py35_0
xz                        5.2.2                         1
yaml                      0.1.6                         0
zlib                      1.2.8                         3
```

**Comment by subesokun** [comment]
After debugging I've noticed that the bug is caused by a wrong usage of the `requests` library. 

https://github.com/spotify/luigi/blob/a69dffb48dd6401261e764ffbf06c2d3b9ad4cd3/luigi/rpc.py#L85 You're passing in the GET request query arguments via `data=body` but a GET request has no body. Instead you should use `params=body`. Hence the server doesn't receive any query arguments and is failing with a quite ugly 500 internal server error.

**Comment by Tarrasch** [comment]
Would you mind creating a PR fixing this @subesokun ?

**Comment by danielnorberg** [comment]
@Tarrasch We're hitting this issue. Is the master branch ready to release?

**Comment by honnix** [comment]
@Tarrasch Why did we close #2272 ? I think this issue is still valid and needs to be fixed.

**Comment by Tarrasch** [comment]
@honnix, no idea, I've not been involved in this. Feel free to dig into this.

**Comment by fess10** [comment]
@honnix I'm not sure whether the problem can be actually solved by #2772, guess that it needs further investigation.

**Comment by samael500** [comment]
I've got the same case, and change `data=` to `param=` was fixed it.
